### PR TITLE
[ios] Add example for label placement.

### PR DESCRIPTION
**Pull request by fabian-guerra**
Example for label collision placement.

### CODE REVIEW

File: Examples/Examples.h [hunk 1]
```diff
@@ -61,6 +61,7 @@ extern NSString *const MBXExampleThirdPartyVectorStyle;
 extern NSString *const MBXExampleUserLocationAnnotation;
 extern NSString *const MBXExampleUserTrackingModes;
 extern NSString *const MBXExampleWebAPIData;
+extern NSString *const MBXExampleLabelPlacement;
```
Comment (captainbarbosa) [inline Examples/Examples.h]: Please put this under line 40 to keep this in alphabetical order.

File: Examples/ObjectiveC/Headers/LabelPlacementExample.h [hunk 2]
```diff
@@ -0,0 +1,17 @@
+//
+//  LabelPlacementExample.h
+//  Examples
+//
+//  Created by Fabian Guerra Soto on 4/2/19.
+//  Copyright © 2019 Mapbox. All rights reserved.
+//
+
```
Comment (captainbarbosa) [inline Examples/ObjectiveC/Headers/LabelPlacementExample.h]: It is customary for this project to remove the above information.

File: Examples/ObjectiveC/LabelPlacementExample.m [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    [self.view addSubview:self.mapView];
+}
+
+- (void)mapView:(MGLMapView *)mapView didFinishLoadingStyle:(MGLStyle *)style {
+    MGLSymbolStyleLayer *poiLabelLayers = (MGLSymbolStyleLayer *)[self.mapView.style layerWithIdentifier:@"poi-label"];
+    poiLabelLayers.textAnchor = nil;
+    poiLabelLayers.textOffset = nil;
+    poiLabelLayers.symbolPlacement = nil;
+    poiLabelLayers.textOffset = nil;
+//    poiLabelLayers.textVariableAnchor = [NSExpression expressionForConstantValue:@[@(MGLTextAnchorTop), @(MGLTextAnchorRight)]];
```
Comment (captainbarbosa) [inline Examples/ObjectiveC/LabelPlacementExample.m]: If you intend to keep these comments in, can you explain what uncommenting them will do?

File: Examples/ObjectiveC/LabelPlacementExample.m [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    self.mapView.styleURL = [MGLStyle streetsStyleURL];
+    [self.mapView setCenterCoordinate:CLLocationCoordinate2DMake(37.791282, -122.396301)
+                            zoomLevel:15.0
+                             animated:NO];
+    self.mapView.delegate = self;
+    [self.view addSubview:self.mapView];
+}
+
+- (void)mapView:(MGLMapView *)mapView didFinishLoadingStyle:(MGLStyle *)style {
+    MGLSymbolStyleLayer *poiLabelLayers = (MGLSymbolStyleLayer *)[self.mapView.style layerWithIdentifier:@"poi-label"];
```
Comment (captainbarbosa) [inline Examples/ObjectiveC/LabelPlacementExample.m]: Its important to mention here that the `poi-label` layer is specific to Mapbox styles that include the `mapbox-streets-v8` source, which you may want to link to here: https://docs.mapbox.com/vector-tiles/reference/mapbox-streets-v8/#poi_label

File: Examples/ObjectiveC/LabelPlacementExample.m [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    poiLabelLayers.textAnchor = nil;
+    poiLabelLayers.textOffset = nil;
+    poiLabelLayers.symbolPlacement = nil;
+    poiLabelLayers.textOffset = nil;
```
Comment (captainbarbosa) [inline Examples/ObjectiveC/LabelPlacementExample.m]: Can you please explain what the default styling for the `poi-label` looks like? For example, by default it shows a symbol (or "icon") with the text below it, and the modification you are making above will undo that.

File: Examples/ObjectiveC/LabelPlacementExample.m (hunk 3 above)
Comment (fabian-guerra) [reply Examples/ObjectiveC/LabelPlacementExample.m]: This is the code that does the label placement. But I don't have a beta release that contains such code, that's why I commented this code while waiting for beta.2. Once that is released will uncomment.

File: Examples/ObjectiveC/LabelPlacementExample.m (hunk 5 above)
Comment (fabian-guerra) [reply Examples/ObjectiveC/LabelPlacementExample.m]: This refers to label collision, when you turn on the label placement anchors the collision detection re-positions the labels.

Here is how it looks:
https://github.com/mapbox/mapbox-gl-js/pull/7596#issue-231698226

**Pull request merged**